src/Proto.Mailbox/Mailbox.cs

```diff
@@ -115,10 +115,9 @@ private Task RunAsync()
                     _stats[i].MailboxEmpty();
                 }
             }
-            return Task.FromResult(0);
         }
 
-        private bool ProcessMessages()
+        private async Task<bool> ProcessMessages()
```

Comment:
I think we could handle the Task returned by EscalateFailure the same way as the one returned when invoking the message. Continue if complete, otherwise bail out and reschedule